pkg/machine/ocipull/ociartifact.go

```diff
@@ -92,8 +109,17 @@ func NewOCIArtifactPull(ctx context.Context, dirs *define.MachineDirs, endpoint
 
 	cache := false
 	if endpoint == "" {
-		imageName := artifactImageName
-		endpoint = fmt.Sprintf("docker://%s/%s/%s:%s", artifactRegistry, artifactRepo, imageName, artifactVersion.majorMinor())
+		// Get machine image from configuration
+		machineImage, err := getMachineImageConfig()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get machine image configuration: %w", err)
+		}
+
+		if _, err := validateMachineImageConfig(machineImage); err != nil {
+			return nil, fmt.Errorf("machine image ref could be misconfigured in containers.conf: %w", err)
+		}
```

Comment:
I don't see why we would need an extra function to validate anything here, the pull logic below will fail if this is invalid no matter what so this adds just overhead